### DISCUSSION TITLE
How do we make valent open at startup as a background process

### DISCUSSION THREAD

**Discussion by NaiveInvestigator** [andyholmes/valent#627] [Help] [answered]
As stated above. I can make valent start by starting up the GUI and then the background process starts. But is there any way to do directly instead of calling the GUI?

**Comment by andyholmes** [comment]
If you install the Flatpak, Valent will install an XDG autostart file the first time you open the main window (and give it permission).

If you installed it from a package, then it should have been installed to the appropriate system directory (if the package is setup correctly).

Or from the command line, I guess

```sh
flatpak run --command=valent ca.andyholmes.Valent --gapplication-service
```

**Comment by NaiveInvestigator** [comment]
I installed valent from the AUR, not the flatpak version. And i use hyprland as my DE so i need to manually add everything i want to add at startup. Where could we get the XDG autostart file?

Edit: I mean the xdg autostart file location. If you could give the command line to start the service would be really helpful!

Edit - 2: Or is there a systemd command which starts up the service automatically?

**Reply by andyholmes** [reply, marked as answer]
> Where could we get the XDG autostart file?

It should be installed as part of the package, so it depends on the prefix set at build time and so on.

> [...] If you could give the command line to start the service would be really helpful!

It's the same command, just not run in a flatpak:

```
valent --gapplication-service
```

**Reply by NaiveInvestigator** [reply]
TYSM!!! This did the trick! 

For any new hyprland users, adding this to your hyprland.conf file should do the trick:

```
exec-once = valent --gapplication-service
```

Thank you again for the help and the GREAT work on valent ;-;. I never used flatpaks so i didnt understand ;-;.  I was stuck on this for the past few days, I'm new to linux in general lol.

Hopefully someday i can figure out how to sync the keyboards automatically.

Please have a pleasant day or night :>>
